pkg/asset/manifests/azure/cluster.go

```diff
@@ -234,9 +240,26 @@ func GenerateClusterAssets(installConfig *installconfig.InstallConfig, clusterID
 			},
 		},
 	}
-	azureCluster.SetGroupVersionKind(capz.GroupVersion.WithKind("AzureCluster"))
+
+	// We are maintaining a fork of CAPZ for azurestack. The only API difference
+	// is the ARMEndpoint field, so we can use the CAPZ cluster object, and if
+	// running on ASH convert to the fork API, and add the field.
+	var cluster client.Object
+	cluster = azureCluster
+	if !strings.EqualFold(azEnv, string(azure.StackCloud)) {
+		azureCluster.SetGroupVersionKind(capz.GroupVersion.WithKind("AzureCluster"))
+	} else {
+		var ashCluster capzash.AzureCluster
+		if err := deepCopy(azureCluster, &ashCluster); err != nil {
+			return nil, fmt.Errorf("failed to convert azureCluster to azure-stack cluster: %w", err)
+		}
+		ashCluster.Spec.ARMEndpoint = session.Environment.ServiceManagementEndpoint
+		ashCluster.SetGroupVersionKind(capzash.GroupVersion.WithKind("AzureCluster"))
+		cluster = &ashCluster
+	}
```

Comment:
This is the best option I could come up with on how to support setting the additional `ARMEndpoint` field that is in the CAPZASH fork--but not CAPZ. Feedback/ideas welcome.